### DISCUSSION TITLE
JSLT for Nested Loops

### DISCUSSION THREAD

**Discussion by hemakumarg89** [schibsted/jslt#196] [Q&A]
Hi Team,
Pardon if it is a basic question as I am new to JSLT.

I am trying to write a JSLTransform which lowercases all the json-attributes (not the values) including the nested structures. I could not managed to  do that myself and getting below error. Could not find much documentation as well. Please could you guide me ? Attached the jslt file
[SampleJSLT.txt](https://github.com/schibsted/jslt/files/6593259/SampleJSLT.txt)

`Exception in thread "main" com.schibsted.spt.data.jslt.JsltException: Parse error: Encountered " <IDENT> "toLower "" at line 22, column 1.
Was expecting one of:
    <EOF>
    "or" ...
    "and" ...
    "==" ...
    "!=" ...
    ">=" ...
    ">" ...
    "<" ...
    "<=" ...
    "+" ...
    "-" ...
    "*" ...
    "/" ...
    "|" ...
     at /data/apps/hub/utils/JSLTFiles/iviResponse.jslt:21:5
        at com.schibsted.spt.data.jslt.parser.ParserImpl.compileExpression(ParserImpl.java:61)
        at com.schibsted.spt.data.jslt.Parser.compile(Parser.java:226)
        at com.schibsted.spt.data.jslt.Parser.compile(Parser.java:62)
        at com.schibsted.spt.data.jslt.Parser.compile(Parser.java:51)
        at com.lnrs.jslt.JSLTransformWorkout.main(JSLTransformWorkout.java:23)`

**Comment by larsga** [comment]
Your function is basically:

```
def toLower(node)
  [ ... ]
  { ... }
```

How is that going to work? I mean, do you want to return an array or an object? You can't do both.

Maybe at the top level of your function you want to use an if to check `is-array($node)`, then `is-object($node)` and finally just return `$node` if it's neither?

**Reply by hemakumarg89** [reply]
Hi, Thanks for the response. As I mentioned earlier I am in the learning phase.
Could you be kindly guide me how this could be rectified please . Thanks

**Reply by larsga** [reply]
Something like this maybe?

```
def toLower(node)
  if (is-array($node))
    [ ... ]
  else if (is-object($node))
    { ... }
  else
    $node
```

**Reply by hemakumarg89** [reply]
That works. Thanks alot. Appreciate the immediate response
